src/common/src/types/interval.rs

```diff
@@ -1007,6 +1010,78 @@ impl Interval {
         };
         format!("P{years}Y{months}M{days}DT{hours}H{minutes}M{seconds}{fract_str}S")
     }
+
+    pub fn from_iso_8601(s: &str) -> Result<Self> {
+        // ISO pattern - PnYnMnDTnHnMnS
+        static ISO_8601_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+            Regex::new(r"P([0-9]+)Y([0-9]+)M([0-9]+)DT([0-9]+)H([0-9]+)M([0-9]+(?:\.[0-9]+)?)S")
+                .unwrap()
+        });
+        let caps = ISO_8601_REGEX
+            .captures(s)
+            .ok_or_else(|| ErrorCode::InvalidInputSyntax(format!("Invalid interval: {}", s)))?;
+        // `caps.get` is safe to unwrap, since `caps` is `Some`
+        let years: i32 = caps
+            .get(1)
+            .unwrap()
+            .as_str()
+            .parse()
+            .map_err(|_| ErrorCode::InvalidInputSyntax(format!("Invalid interval: {}", s)))?;
+        let months: i32 = caps
+            .get(2)
+            .unwrap()
+            .as_str()
+            .parse()
+            .map_err(|_| ErrorCode::InvalidInputSyntax(format!("Invalid interval: {}", s)))?;
+        let days = caps
+            .get(3)
+            .unwrap()
+            .as_str()
+            .parse()
+            .map_err(|_| ErrorCode::InvalidInputSyntax(format!("Invalid interval: {}", s)))?;
+        let hours: i64 = caps
+            .get(4)
+            .unwrap()
+            .as_str()
+            .parse()
+            .map_err(|_| ErrorCode::InvalidInputSyntax(format!("Invalid interval: {}", s)))?;
+        let minutes: i64 = caps
+            .get(5)
+            .unwrap()
+            .as_str()
+            .parse()
+            .map_err(|_| ErrorCode::InvalidInputSyntax(format!("Invalid interval: {}", s)))?;
+        // usecs = sec * 1000000, use decimal to be exact
+        let usecs: i64 = (Decimal::from_str_exact(caps.get(5).unwrap().as_str())
```

Comment:
`caps[6]` instead of `caps[5]`? Seems not covered by tests

Reply:
ohhhh my mistake!
added a stronger case :)